src/python/national_voter_file/us_states/ut/transformer.py

```diff
@@ -0,0 +1,280 @@
+import csv
+import io
+import os
+import re
+import sys
+
+from national_voter_file.transformers.base import (DATA_DIR,
+                                                   BasePreparer,
+                                                   BaseTransformer)
+import usaddress
+
+__all__ = ['default_file', 'StatePreparer', 'StateTransformer']
+
+default_file = 'voters.txt'
+
+class StatePreparer(BasePreparer):
+
+    state_path = 'ut' # Two letter code for state
+    state_name='Utah' # Name of state with no spaces. Use CamelCase
+    sep=',' # The character used to delimit records
+
+    def __init__(self, input_path, *args):
+        super(StatePreparer, self).__init__(input_path, *args)
+
+        if not self.transformer:
+            self.transformer = StateTransformer()
+
+    def process(self):
+            fp = open(self.input_path, 'r', encoding='utf-8-sig')
+
+            reader = self.dict_iterator(fp)
+            for row in reader:
+                yield row
+
+class StateTransformer(BaseTransformer):
+    date_format='%m/%d/%Y' # The format used for dates
+    input_fields = None # This can be a list of column names for the input file.
+                        # Use None if the file has headers
+
+    col_map = {
+        'TITLE': None,
+        'FIRST_NAME': 'First Name',
+        'LAST_NAME': 'Last Name',
+        'MIDDLE_NAME': 'Middle Name',
+        'NAME_SUFFIX': 'Name Suffix',
+        'GENDER': None,
+        'RACE': None,
+        'BIRTH_STATE': None,
+        'LANGUAGE_CHOICE': None,
+        'EMAIL': None,
+        'PHONE': 'Phone',
+        'DO_NOT_CALL_STATUS': None,
+        'ABSENTEE_TYPE': 'Absentee',
+        'CONGRESSIONAL_DIST': 'Congressional',
+        'UPPER_HOUSE_DIST': 'State Senate',
+        'LOWER_HOUSE_DIST': 'State House',
+        'SCHOOL_BOARD_DIST': 'State Schoolboard',
+        'COUNTY_BOARD_DIST': None, # TODO: what is this?
+        'COUNTYCODE': 'County ID',
+        'COUNTY_VOTER_REF': None,
+        'PRECINCT': 'Precinct',
+        'PRECINCT_SPLIT': None,
+        'STATE_VOTER_REF': 'Voter ID',
+    }
+
+    ut_party_map = {
+        'Republican': 'REP',
+        'Unaffiliated': 'UN',
+        'Democratic': 'DEM',
+        'Libertarian': 'LIB',
+        'Independent American': 'AI',
+        'Constitution': 'AMC',
+        #'American': '',
```

Comment:
I pulled all distinct values for the Party column and mapped the ones I could. The remaining (commented) values were ones that appeared in the data file but which I did not know how to map.